stdlib/shutil.pyi

```diff
@@ -75,7 +75,7 @@ def chown(path: StrOrBytesPath, user: str | int, group: str | int) -> None: ...
 
 if sys.version_info >= (3, 8):
     @overload
-    def which(cmd: StrPath, mode: int = ..., path: StrPath | None = ...) -> str | None: ...
+    def which(cmd: StrPath, mode: int = ..., path: StrPath | None = ...) -> StrPath | None: ...
```

Comment:
Returning `StrPath` introduces new errors, because it is a union type. If I pass in both `cmd` and `path` as strings, the result will be a string or None, and the return type shouldn't annoyingly be `StrPath | None`. Please add a new overload to cover this use case.

Reply:
Actually, a `TypeVar` would be the best way to say "can return a string, None, or the type of cmd". This way we don't need a new overload.